### PR TITLE
Deprecate implicit includeSCPI declaration

### LINKED ISSUE #1000: Deprecate implicit includeSCPI definition for mixin based instruments
The transition from our current setup with `Instrument` including SCPI commands per default to a mixin based process (#804 , #905 ) is difficult as `includeSCPI` is True on default, but the future default will be without SCPI.

In order to facilitate the transition, I propose the following:

1. Change the default value to `None`
2. In the init, `None` will be converted to `True`, but will raise a deprecation warning.
3. We could add a test for new instruments that they have `includeSCPI` defined.

That change should prompt people to include a definition into their driver (those, which are in pymeasure and those which they use privately).

What do you think @bilderbuchi , @CasperSchippers , @msmttchr ?

### PR DESCRIPTION
**Pull request by BenediktBurger**
In preparation for #804, #905, this PR deprecates not declaring `includeSCPI` for enabling SCPI functions.
That allows later to switch to SCPI mixin class. Closes #1000 

Does not break working code: If it is  not declared, only a warning is raised and SCPI is enabled (as previously).

An "all_instruments" test checks for new instruments to declare `includeSCPI`.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -4,6 +4,7 @@ Upcoming version
 Deprecated features
 -------------------
 - Replaced :code:`directory_input` keyword-argument of :code:`ManagedWindowBase` by :code:`enable_file_input` (@CasperSchippers, #964)
+- Deprecated to not specify :code:`incldueSCPI` for an instrument which uses SCPI (@BenediktBurger, #1007)
```
Comment (bilderbuchi) [inline CHANGES.rst:7]:
```suggestion
- Deprecated to not specify :code:`includeSCPI` for an instrument which uses SCPI (@BenediktBurger, #1007)
```

File: tests/instruments/test_all_instruments.py [hunk 2]
```diff
@@ -215,6 +215,91 @@ def find_devices_in_module(module):
     "ChannelAWG",
     "ChannelAFG",
 ]
+# Instruments which do not YET define `includeSCPI` explicitly
+grandfathered_includeSCPI_instruments = [
```
Comment (bilderbuchi) [inline tests/instruments/test_all_instruments.py:219]: do you plan a follow-up PR that converts those instruments to the new usage? That would be good!

Comment (BenediktBurger) [reply tests/instruments/test_all_instruments.py:219]: I cannot do that work alone, as I do not know, which instrument is Scpi compliant.

We have to rely on users submitting that information (they will get a warning).
Later (in the restructuring), we can add a "scpiUnknownMixin" (see my other pr) for these instruments.
That will keep the code base in a working state, but will also mark these instruments as maybe not Scpi compliant.

Tdlr: no large pr planned, but in small pieces over time.

Comment (bilderbuchi) [reply tests/instruments/test_all_instruments.py:219]: I see (I think). So far, if not specified, we assume that SCPI inclusion is true, so assuming that the instruments in the codebase are implemented correctly, we could just carry that current behaviour over into explicit settings? Or do you want to take this opportunity to confirm for all those instruments if SCPI compatibility is _really_ appropriate?

Comment (BenediktBurger) [reply tests/instruments/test_all_instruments.py:219]: I don't mind shipping both PRs together.

We should raise some warning, though, that Scpi compliance is not known in order to inform users.
That will trigger people to add the information, but won't stop code from working.

File: CHANGES.rst (hunk 1 above)
Comment (BenediktBurger) [reply CHANGES.rst]: I don't understand your message, therefore, I made a new suggestion, which is, hopefully, easier to understand.

Comment (bilderbuchi) [reply CHANGES.rst]: No problem! However
```
:code:`incldueSCPI` obligarory for
```
-> 
```
:code:`includeSCPI` obligatory for
```

Comment (BenediktBurger) [reply CHANGES.rst]: Done. Thanks.

File: tests/instruments/test_all_instruments.py (hunk 2 above)
Comment (BenediktBurger) [reply tests/instruments/test_all_instruments.py:219]: > We should raise some warning, though, that Scpi compliance is not known in order to inform users.
That will trigger people to add the information, but won't stop code from working.

The deprecation warning is that warning.

File: CHANGES.rst [hunk 3]
```diff
@@ -4,7 +4,7 @@ Upcoming version
 Deprecated features
 -------------------
 - Replaced :code:`directory_input` keyword-argument of :code:`ManagedWindowBase` by :code:`enable_file_input` (@CasperSchippers, #964)
-- Make parameter :code:`incldueSCPI` obligarory for all instruments, even those which use SCPI (@BenediktBurger, #1007)
+- Make parameter :code:`incldueSCPI` obligatory for all instruments, even those which use SCPI (@BenediktBurger, #1007)
```
Comment (bilderbuchi) [inline CHANGES.rst:7]:
```suggestion
- Make parameter :code:`includeSCPI` obligatory for all instruments, even those which use SCPI (@BenediktBurger, #1007)
```

**Comment by BenediktBurger** [comment]
I did all the changes, anything left, @bilderbuchi?

**Review by bilderbuchi — APPROVED** [review]
LGTM, thanks!

**Pull request merged**